src/modules/kvs/kvs.c

```diff
@@ -2850,13 +2945,26 @@ int mod_main (flux_t *h, int argc, char **argv)
     if (ctx->rank == 0) {
         struct kvsroot *root;
         char rootref[BLOBREF_MAX_STRING_SIZE];
+        double timestamp = 0.;
         uint32_t owner = getuid ();
 
         /* Look for a checkpoint and use it if found.
          * Otherwise start the primary root namespace with an empty directory.
          */
-        if (checkpoint_get (h, "kvs-primary", rootref, sizeof (rootref)) == 0)
-            flux_log (h, LOG_INFO, "restored kvs-primary from checkpoint");
+        if (checkpoint_get (h, "kvs-primary",
+                            rootref, sizeof (rootref), &timestamp) == 0) {
+            char datestr[128];
+            time_t sec = timestamp;
+            struct tm tm;
+            if (timestamp > 0.) {
+                gmtime_r (&sec, &tm);
+                strftime (datestr, sizeof (datestr), "%FT%T", &tm);
+            }
+            else
+                snprintf (datestr, sizeof (datestr), "N/A");
+            flux_log (h, LOG_INFO,
+                      "restored kvs-primary from checkpoint on %s", datestr);
```

Comment:
This should probably be in its own function, if not folded into `checkpoint_get()`.